### DISCUSSION TITLE
Rename of vsg::UniformBinding to vsg::DescriptorBinding and associated method renaming/deprecation

### DISCUSSION THREAD

**Discussion by robertosfield** [vsg-dev/VulkanSceneGraph#964] [General]
Hi All,

I have just merged changes to VSG master that renames vsg::UniformBinding to vsg::DescriptorBinding and also renames a collection of enableUniform(..)/assignUniform(..) methods in vsg;;DescriptorConfiguration and vsg::GraphicsPipelineConfigurator to enbleDesciptor/enableDescriptor(..). The original methods have been [[deprecated]] so you'll now get warnings to rename them when you build against VSG master or the next stable release 1.0.10.  

The changes to the VSG are:

https://github.com/vsg-dev/VulkanSceneGraph/pull/963

The deprecated methods were used in vsgXchange, vsgPoints and vsgExamples so I have merged changes to them to update them to the lastest API.  Changes are just renames of methods/classes so it's not a major task:

https://github.com/vsg-dev/vsgXchange/pull/169
https://github.com/vsg-dev/vsgPoints/pull/14
https://github.com/vsg-dev/vsgExamples/pull/262

These changes were made to resolve some clumsy naming of class and methods which was highlighted when @theodoregoetz  recently tried to use ShaderSet/GraphicsPipelineConfigurator with a DescruptorBuffer set up as storage buffer type rather than a uniform buffer. type that these classes had been designed around.  The new naming is more general purpose.  

Thanks to @theodoregoetz helping out with these changes and making fixes to the storage buffer support.

**Comment by theodoregoetz** [comment]
I noticed that this PR renames the `vsg::UniformBinding` struct and `vsg::ShaderSet::uniformBindings` public member without providing a deprecated symbol. You may consider adding these back in to `ShaderSet.h`:

```
class ShaderSet {
        std::vector<DescriptorBinding>& uniformBindings = descriptorBindings;
}
```

```
using UniformBinding = DescriptorBinding;
```

**Reply by robertosfield** [reply]
I didn't add it because I didn't hit any compatibility issues with it not being defined.  If there is an issue with compatibility then I'll add it, but will avoid unless we actually need it.
